File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int i = 0;
        int j = nums.length - 1;
        for( ; i < nums.length && i <= j ; i++){
            if(nums[i] == val){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                j--; 
                i--; // to return back to the previous index just in case what was swappeed was the val.
            }
        }
        return i;
    }
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, ptr = 0, len(nums) - 1
        while i <= ptr:
            num = nums[i]
            if num == val:
                while nums[ptr] == val:
                    if i == ptr:
                        return ptr
                    ptr -= 1
                nums[i], nums[ptr] = nums[ptr], nums[i]
            i += 1
        return i
            
        